src/Components/WebAssembly/Authentication.Msal/src/Interop/AuthenticationService.ts

```diff
@@ -0,0 +1,363 @@
+import * as Msal from 'msal';
+import { StringDict } from 'msal/lib-commonjs/MsalTypes';
+import { ClientAuthErrorMessage } from 'msal/lib-commonjs/error/ClientAuthError';
+
+interface AccessTokenRequestOptions {
+    scopes: string[];
+    returnUrl: string;
+}
```

Comment:
It looks like there's quite a lot of duplication here with our other authentication service .ts files. I know this has an advantage of being self-contained, whereas it also has the disadvantage of duplicating the definition of our authentication service contract.

At some point in the future, do you think we should consider publishing something to NPM defining the service contract as a TypeScript abstract class or interface?

We could even consider doing that without publishing it to NPM by just having a shared `ServiceContract.ts` file in our repo that gets imported by all the things that are using it.

Not saying this is essential for this PR, just want to get your thoughts on the relevance of this.

Reply:
Looking at it more, it would be quite nice to factor out lines 5-51 into a separate `.ts` file that's shared across all our authentication service implementations. What do you think?

Reply:
It'll be hard to make typescript consume it I think. I'll give it a quick try, but I'll leave it as is if I can't make it work.

Reply:
I think you can freely do `import ... from '../../../something`

Or if you want to be fancier, you can do `import ... from 'ourSharedSources/something` and define `ourSharedSources` in `tsconfig.json` (see [path mapping docs](https://www.typescriptlang.org/docs/handbook/module-resolution.html#path-mapping)).

Not a big deal if you choose not to do it.